### ISSUE TITLE
StompBadStateException after call unsubscribe function.

### ISSUE THREAD

**Issue by TaranenkoDenis** [blackhorse-one/stomp_dart_client#54]
Hello. 

There are many sockets in our application and we ran into the problem when trying to call StompUnsubscribe: 
`StompBadStateException error occurs: The StompHandler has no active connection or the connection was unexpectedly closed. `
This happens many times and we don't know how to fix it. It is important to note that this does NOT happen every time we try to call StompUnsubscribe.

Lib version: `stomp_dart_client: ^0.4.2`

flutter doctor

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] Connected device (2 available)

• No issues found!
```

**Comment by KammererTob** [comment]
Hey.

This usally means that by the time you are trying to unsubscribe (thus sending a UNSUBSCRIBE frame to the server), the connection is already closed. Most of the time you probably are fine by just checking if stomp is still connected before using the unsubscribe function. Since Stomp does not automatically keep your subscriptions or resubscribe, the subscription is no longer active anyways when stomp is no longer connected. If you believe that this also happens when the connection is still active and good then please let me know.

Thanks!

**Comment by TaranenkoDenis** [comment]
I made the next check before unsubscribing:

```
if (stompClient?.connected == true) {
      _log.d('Stomp client is connected, need to unsubscribe.');
      unsubscribeFromRawData(id);
    } else {
      _log.d('Stomp client is not connected.');
    }
```

And the exception still appears.

```
flutter: 11:26:09.670 🐛 [BaseStompWs] Stomp client is connected, need to unsubscribe.
flutter: 11:26:09.671 🐛 [BaseStompWs] Unsubscribe 
flutter: 11:26:09.672 ⛔ [BaseStompWs] Stomp unsucbscription error:
flutter: ⛔ StompBadStateException: The StompHandler has no active connection or the connection was unexpectedly closed.
```

**Comment by KammererTob** [comment]
Hey,

do you happen to have a reliable reproduction? Or does this only happen poradically?

Thanks!

**Comment by TaranenkoDenis** [comment]
Hello.

Unfortunately, we do not have a clear algorithm for reproducing the error. It is just sometimes reproduced and sometimes not. 

Thanks.

**Comment by KammererTob** [comment]
Hey,

i am currently a bit a busy. I will try to fix this issue as soon as i find the time to build a reproduction. Just FYI

**Comment by KammererTob** [comment]
Hey.

I've just released 0.4.3 which should fix this issue. The issue was that the returned unsubscribe function still had the old state in its scope and thus could try to transmit messages thru an inactive sink, although the user sees the socket as still connected. It now checks the connection state before transmitting the unsubscribe frame. If you still have this issue please feel free to reopen.
